include/base/libmesh_logging.h

```diff
@@ -45,6 +50,44 @@ class PerfLog;
 extern PerfLog perflog;
 
 
+/**
+ * Used for logging something that naturally lasts as long as some
+ * enclosing scope, such as the current function.  Makes it very easy
+ * to handle multiple return scenarios, since the event is popped in
+ * the destructor.  Should not be used directly, instead use the
+ * LOG_SCOPE macro, which resolves to nothing at compile time if
+ * logging is disabled.
+ *
+ * \author John Peterson
+ * \date 2016
+ */
+struct PerfItem
+{
+  PerfItem(const char * label,
+           const char * header,
+           bool enabled=true) :
+    _label(label),
+    _header(header),
+    _enabled(enabled)
+  {
+    if (_enabled)
+      libMesh::perflog.push(label, header);
```

Comment:
Yeah, I was thinking about implementing that as another constructor argument that defaults to `libMesh::perflog`, but in the end decided it wasn't worth the extra macro name, or introducing variadic macros for.  This could definitely be added in the future, though.  Some of the examples create their own PerfLogs and could make use of it, but none of them have the nasty multiple-returns-requiring-matching-pops-for-correctness case.